store/src/java/com/zimbra/cs/util/JMSession.java

```diff
@@ -347,10 +351,38 @@ private static Properties getJavaMailSessionProperties(Server server, Domain dom
         if (domain != null) {
             props.setProperty("mail.host", domain.getName());
         }
-
+        
         return props;
     }
 
+    private static Properties configureStartTls (Properties props, Server server, Domain domain) {
+        SmtpStartTlsMode startTlsMode = null;
+        if (domain != null) {
+            startTlsMode = domain.getSmtpStartTlsMode();
+        }
+        if (startTlsMode == null) {
+            startTlsMode = server.getSmtpStartTlsMode();
+        }
+
+        if (startTlsMode.isOff()) {
+            props.setProperty("mail.smtp.starttls.enable", "false");
+        } else if (startTlsMode.isOnly()) {
+            props.setProperty("mail.smtp.starttls.enable", "true");
+            props.setProperty("mail.smtp.starttls.required", "true");
+            props.setProperty("mail.smtp.ssl.trust", "*");
+        } else {
+            if (!startTlsMode.isOn()) {
+                // Should not be reached here
+                ZimbraLog.smtp.warn("invalid value configured for %s. fallback to \"on\".", Provisioning.A_zimbraSmtpStartTlsMode);
+            }
+            props.setProperty("mail.smtp.starttls.enable", "true");
+            props.setProperty("mail.smtp.starttls.required", "false");
+            props.setProperty("mail.smtp.ssl.trust", "*");
```

Comment:
I believe this is disabling certificate verification, right? I'm not sure that should be our default setting, although this is certainly something that could be configurable.

Reply:
The code only talk to Zimbra MTA (specified by zimbraSmtpHostname), I didn't think we need to verify the server cert on TLS. However I remove these lines to set mail.smtp.ssl.trust to avoid hard coding.